ripple/src/main/webapp/script.js

```diff
@@ -22,17 +29,65 @@ function fetchBlobstoreUploadUrl(formId, fileId, webUrl) {
       });
 }
 
+/* Serves blob by parsing blobKey from parameter. If no parameter, serves stored blob. */
+function getBlobKey(uid, blobKey) {
+  var newBlobKey = getParameterByName('blob-key');
+  console.log("Parameter blobKey: " + newBlobKey);
+  // New image uploaded. Store blobkey in firestore.
+  if (newBlobKey != null) {
+    console.log("New image. Blobkey parameter: ", newBlobKey);
+    db.collection("users").doc(uid).update({
+      blobKey: newBlobKey,
+    })
+        .then(function() {
+        console.log("Document successfully updated!");
+        })
+        .catch(function(error) {
+            // The document probably doesn't exist.
+            console.error("Error updating document: ", error);
+        });
+    serveBlob(newBlobKey, "nav-bar-avatar");
+    serveBlob(newBlobKey, "profile-image");
+  } else {
+    console.log("firestore blobKey: " + blobKey);
+    serveBlob(blobKey, "nav-bar-avatar");
+    serveBlob(blobKey, "profile-image");
+  }
+}
+
+/* Read parameter in URL of window */
+function getParameterByName(name) {
+  console.log('called getParameterByName()');
+  url = window.location.href;
+  name = name.replace(/[\[\]]/g, '\\$&');
```

Comment:
Add a comment here explaining what the regex is trying to accomplish.

Reply:
Fixed